APE6.rst

```diff
@@ -0,0 +1,499 @@
+  # - {name: b, unit: km, datatype: int64, description: This is column b}
+  a b
+  001 2
+  004 3
+
+We see that header starts with a header line and YAML block marker to identify
+the format and the beginning of the data block.  After that comes the YAML data
+structure with column definitions.  Finally the column names and data values
+are included in CSV format with a space delimiter.
+
+Now we can read back the table and see that it has survived the round-trip
+to a text file::
+
+  >>> t2 = Table.read('example.ecsv', format='ascii.ecsv')
+  >>> print(t2)
+    a    b 
+  m / s  km
+  ----- ---
+    001   2
+    004   3
+
+To illustrate the full features of ECSV we create a table with meta data
+(keywords and comments) as well as more detailed column meta data::
+
+  >>> from astropy.utils import OrderedDict
+  >>> t = Table([[1, 4], [2, 3]], names=['a', 'b'])
+  >>> t.meta['keywords'] = OrderedDict([('z_key1', 'val1'), ('a_key2', 'val2')])
+  >>> t.meta['comments'] = ['Comment 1', 'Comment 2', 'Comment 3']
+  >>> t['a'].unit = u.m / u.s
+  >>> t['a'].format = '%5.2f'
+  >>> t['a'].description = 'Column A'
+  >>> t['b'].meta = dict(column_meta={'a':1, 'b': 2})
+
+Now we write the table to standard out::
+
+  >>> import sys
+  >>> t.write(sys.stdout, format='ascii.ecsv')
+  # %ECSV 0.9
+  # ---
+  # datatype:
+  # - {name: a, unit: m / s, type: int64, format: '%5.2f', description: Column A}
+  # - name: b
+  #   datatype: int64
+  #   meta:
+  #     column_meta: {a: 1, b: 2}
+  # meta: !!omap
+  # - keywords: !!omap
+  #   - {z_key1: val1}
+  #   - {a_key2: val2}
+  # - comments: [Comment 1, Comment 2, Comment 3]
+  a b
+  1.00 2
+  4.00 3
+
+In this case there are Ordered dictionary elements which must be preserved
+during serialization.  This is done via the ``!!omap`` element tag.  The
+reference Python implementation in Astropy uses custom Loader and Dumper
+classes to allow serialization to and from the Python ``OrderedDict`` class.
+Other implementations must likewise use an ordered mapping when reading and the
+``!!omap`` tag when writing for ordered mappings in the data structure.
+
+In addition, the reference Python implementation outputs the column attributes
+in the order ``'name'``, ``'unit'``, ``'datatype'``, ``'format'``,
+``'description'``, and ``'meta'``.  This is not a ECSV requirement but is
+recommended for human accessibility.
+
+Header details
+^^^^^^^^^^^^^^^^
+
+An ECSV file begins with the the table header which contains the
+necessary information to define the table columns and metadata.  This
+is expressed as a YAML-encoded data structure which has a small set of
+required keywords and standard specifiers.
+
+Each line of the YAML-encoded data structure must start with the two
+characters ``# `` (hash followed by space) to indicate the presence of
+header content.  The first line which does not start with ``#``
+signifies the end of the header.  Subsequent lines starting with ``#``
+are treated as file comment lines.
```

Comment:
@PaulKuin - do you want to read QDP files to be readable with astropy?  That's a different question. In that case the comment character is already configurable via the `basic` reader, but right now we don't support trailing line comments.  It would be quite straightforward though to make a new `qdp` reader that works with qdp files.